addons/payment/wizards/payment_link_wizard.py

```diff
@@ -46,7 +46,7 @@ def default_get(self, fields):
 
     @api.onchange('amount', 'description')
     def _onchange_amount(self):
-        if self.amount_max < self.amount:
+        if float_compare(self.amount_max, self.amount, precision_rounding=self.currency_id.rounding) == -1:
```

Comment:
`currency_id` is not required, so this can crash.

```suggestion
        if float_compare(self.amount_max, self.amount, precision_rounding=self.currency_id.rounding or 0.01) == -1:
```